grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -136,26 +136,26 @@ static void attachGeneratedRelations(Representation currentHal, Map.Entry<VarNam
     private static void attachSingleGeneratedRelation(Representation currentHal, Concept currentVar, Concept otherVar, Map<VarName, String> roleTypes, VarName currentVarName, VarName otherVarName, Optional<TypeName> relationType, String keyspace) {
         ConceptId currentID = currentVar.getId();
 
-        String firstID;
-        String secondID;
+        ConceptId firstID;
+        ConceptId secondID;
         String firstRole;
         String secondRole;
 
         if (currentID.compareTo(otherVar.getId()) > 0) {
-            firstID = currentID.getValue();
-            secondID = otherVar.getId().getValue();
+            firstID = currentID;
+            secondID = otherVar.getId();
             firstRole = (roleTypes.get(currentVarName).equals(HAS_ROLE_EDGE)) ? "" : roleTypes.get(currentVarName) + ":";
             secondRole = (roleTypes.get(otherVarName).equals(HAS_ROLE_EDGE)) ? "" : roleTypes.get(otherVarName) + ":";
         } else {
-            firstID = otherVar.getId().getValue();
-            secondID = currentID.getValue();
+            firstID = otherVar.getId();
+            secondID = currentID;
             secondRole = (roleTypes.get(currentVarName).equals(HAS_ROLE_EDGE)) ? "" : roleTypes.get(currentVarName) + ":";
             firstRole = (roleTypes.get(otherVarName).equals(HAS_ROLE_EDGE)) ? "" : roleTypes.get(otherVarName) + ":";
         }
 
-        String isaString = (relationType.isPresent()) ? "isa " + relationType.map(StringConverter::typeNameToString) : "";
-        String assertionID = String.format(ASSERTION_URL, keyspace, firstID, secondID, firstRole, secondRole, isaString);
-        currentHal.withRepresentation(roleTypes.get(currentVarName), new HALGeneratedRelation().getNewGeneratedRelation(firstID, secondID, assertionID, relationType));
+        String isaString = (relationType.isPresent()) ? "isa " + StringConverter.typeNameToString(relationType.get()) : "";
+        String assertionID = String.format(ASSERTION_URL, keyspace, firstID, secondID, firstRole, secondRole,isaString);
+        currentHal.withRepresentation(roleTypes.get(currentVarName), new HALGeneratedRelation().getNewGeneratedRelation(firstID.getValue(),secondID.getValue(),assertionID, relationType));
```

Comment:
Is there any reason `getNewGeneratedRelation ` cannot take in the `ConceptID` as well?